packages/mui-material/src/Radio/Radio.js

```diff
@@ -121,11 +122,26 @@ const Radio = React.forwardRef(function Radio(inProps, ref) {
     onChange: onChangeProp,
     size = 'medium',
     className,
+    disabled: disabledProp,
     disableRipple = false,
     ...other
   } = props;
+
+  const muiFormControl = useFormControl();
+
+  let disabled = disabledProp;
+
+  if (muiFormControl) {
+    if (typeof disabled === 'undefined') {
```

Comment:
Why only read the value if `undefined`?

Reply:
Because the prop should take precedence over what comes form the form field context.

Reply:
Right, my bad 😅 I got confused and thought we were checking `typeof muiFormControl.disabled === 'undefined'`